### ISSUE TITLE
Warnings across multiple Audio discs

### ISSUE THREAD

**Issue by Feathered-Serpent** [superg/redumper#130]
Hey there,

I dumped my audio cds with redumper and on some of them where some warnings, while the cue and bins still have been created. Now I just wanted to see if that is
- a problem with Redumper
- a problem with my discs
- not a problem at all!

Maybe you find some time to look into the files. The warnings are:
```
Disc 1:
warning: TOC / QTOC mismatch, track length (track: AA, LBA: 304944 <=> 304941)

warning: incomplete pre-gap (session: 2, unavailable: 75/150)
warning: lead-in contains non-zero data (session: 2, sectors: 4646/4650)

warning: descramble failed (LBA: [223762 .. 223763])
warning: descramble failed (LBA: [223767 .. 223774])

Disc 2:
warning: lead-out ends with unavailable sector (session: 1)

Disc 3:
warning: TOC / QTOC mismatch, track length (track: A, LBA: 94075 <=> 94072)

warning: incomplete pre-gap (session: 2, unavailable: 75/150)

Disc 4:
warning: incomplete pre-gap (session: 2, unavailable: 74/150)

Disc 5:
warning: TOC / QTOC mismatch, track index 01 (track: 11, LBA: 169055 <=> 169056)
warning: TOC / QTOC mismatch, track index 01 (track: 13, LBA: 204002 <=> 204003)

Disc 6:
warning: offset shift detected, to apply correction please use an option
disc write offset: +1001

warning: incomplete pre-gap (session: 2, unavailable: 74/150)
warning: lead-in contains non-zero data (session: 2, sectors: 76/4650)
warning: lead-out contains non-zero data (session: 2, sectors: 96/97)
warning: lead-out ends with unavailable sector (session: 2)

warning: descramble failed (LBA: [237322 .. 237333])
warning: descramble failed (LBA: [262600 .. 262751])
warning: descramble failed (LBA: [262752 .. 262847])
```
[Disc1.zip](https://github.com/superg/redumper/files/14749244/Disc1.zip)
[Disc2.zip](https://github.com/superg/redumper/files/14749247/Disc2.zip)
[Disc3.zip](https://github.com/superg/redumper/files/14749248/Disc3.zip)
[Disc4.zip](https://github.com/superg/redumper/files/14749249/Disc4.zip)
[Disc5.zip](https://github.com/superg/redumper/files/14749250/Disc5.zip)
[Disc6.zip](https://github.com/superg/redumper/files/14749252/Disc6.zip)

**Comment by superg** [comment]
TOC / QTOC mismatch is a disc authoring problem, redumper is using TOC based split by default.
Usually track AA is leadout and if it's zeroed it's not getting preserved so that can be ignored.
Other TOC / QTOC track mismatches lead to slightly different track start addresses, TOC is favored here but there is an option --force-qtoc to get the other split. This is similar to "(Subs indices)" DIC generated dumps.

`warning: incomplete pre-gap (session: 2, unavailable: 75/150)` can be ignored for multisession disc, it just says that there might be an extra 1 second worth of data that wasn't extracted (as there is no current way to get that data), I plan to reduce verbosity of cases like this because too much noise.

Disc 2: looks like non zero data extends all the way into leadout and plextor can read only ~100 first leadout sectors. To fully dump such disc you should either use trap disc to spoof TOC or use modified ASUS 3.10 firmware from RibShark.

Disc 6: is another mastering issue where disc shifts offset between tracks. From the logs it looks like it detected the wrong offset, I recommend to resplit it with manually specified +1176 offset if you preserved all dump files (redumper split hash info --image-name=ARCADE --force-offset=1176) or just dump again with --force-offset=1176

Descramble failed warnings is expected for bad mastering and usually can be ignored.

I plan some redumper changes that will improve diagnostics in some of your cases but it will take more time.

Hope that explains it all.

**Comment by Feathered-Serpent** [comment]
> if you preserved all dump files
I did, I just removed the .scram, .state and .subcode files from the zips, so they won't become too big for upload.
In fact to archive the original disk, I'll just delete the bin(s) and cue file, as I can always recreate them from the scram data file and then I 7zip the rest of the files to place it onto my nas.

With the information given I can be sure, that there is not a real problem to be seen. 

I do have an ASUS BW-16D1HT drive with the ribshark firmware. Would that mean I wouldn't need the Plextor drive anymore or is it still better than the Asus and I should keep it for dumps of older discs and only use Asus von special cases like that?
